mirrord-config/derive/src/lib.rs

```diff
@@ -0,0 +1,259 @@
+use proc_macro2::{Span, TokenStream};
+use proc_macro2_diagnostics::{Diagnostic, SpanDiagnosticExt};
+use quote::quote;
+use syn::{
+    spanned::Spanned, Data, DataStruct, DeriveInput, Expr, Field, Fields, FieldsNamed,
+    GenericArgument, Ident, Lit, Meta, NestedMeta, PathArguments, Type,
+};
+
+#[derive(Eq, PartialEq, Debug)]
+enum FieldFlags {
+    Unwrap,
+    Nested,
+    Env(Lit),
+    Default(Lit),
+}
+
+/// Parse and create Ident from map_to attribute
+fn map_to_ident(source: &Ident, expr: Option<Expr>) -> Ident {
+    let fallback = Ident::new(&format!("Mapped{}", source), Span::call_site());
+
+    if let Some(Expr::Assign(expr)) = expr {
+        if let (Expr::Path(left_path), Expr::Path(right_path)) = (*expr.left, *expr.right) {
+            if left_path.path.is_ident("map_to") {
+                return right_path
+                    .path
+                    .get_ident()
+                    .cloned()
+                    .expect("map_to value not a valid Ident");
+            }
+        }
+    }
+
+    fallback
+}
+
+/// Parse field attribte to FieldFlags
+fn get_config_flag(meta: &NestedMeta) -> Result<FieldFlags, Diagnostic> {
+    match meta {
+        NestedMeta::Meta(Meta::Path(path)) if path.is_ident("unwrap") => Ok(FieldFlags::Unwrap),
+        NestedMeta::Meta(Meta::Path(path)) if path.is_ident("nested") => Ok(FieldFlags::Nested),
+        NestedMeta::Meta(Meta::NameValue(meta)) if meta.path.is_ident("env") => {
+            Ok(FieldFlags::Env(meta.lit.clone()))
+        }
+        NestedMeta::Meta(Meta::NameValue(meta)) if meta.path.is_ident("default") => {
+            Ok(FieldFlags::Default(meta.lit.clone()))
+        }
+        _ => Err(meta.span().error("unsupported config attribute flag")),
+    }
+}
+
+/// Parse field attribtes to FieldFlags
+fn get_config_flags(meta: Meta) -> Result<Vec<FieldFlags>, Diagnostic> {
+    let result = match meta {
+        Meta::List(list) => {
+            let mut result = Vec::new();
+            for nested in list.nested.iter() {
+                result.push(get_config_flag(nested)?);
+            }
+
+            result
+        }
+        _ => vec![],
+    };
```

Comment:
```suggestion
Meta::List(list) => list.nested.into_iter().flat_map(get_config_flag).collect(),
_ => vec![],
```

ps: requires `get_config_flag` to change as well `(meta: NestedMeta)`.

Reply:
The problem I have with making it a one liner is how to propagate the Err from `get_config_flag` Result

Reply:
Huh, apparently collect has an [implementation](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect)

Reply:
@DmitryDodzin Is this resolved?